### PR TITLE
Fixed logging calls in Split code

### PR DESCRIPTION
**Pull request by shazqadeer**
Closing #484

### CODE REVIEW

**Comment by shazqadeer** [comment]
@RustanLeino: I tested the output manually by using /vcsSplitOnEveryAssert and /vcsCores:4 on my machine on several large Civl examples.

File: Source/VCGeneration/Split.cs [hunk 1]
```diff
@@ -956,7 +959,7 @@ bool ContainsAssert(Block b)
                || c is AssertCmd && splitOnEveryAssert;
       }
       
-      public static List<Split /*!*/> FindManualSplits(Split initialSplit, bool splitOnEveryAssert)
+      public static List<Split /*!*/> FindManualSplits(Split initialSplit, ImplementationRun run, bool splitOnEveryAssert)
```
Comment (keyboardDrummer) [inline Source/VCGeneration/Split.cs:962]: Isn't the run already contained in the `initialSplit` argument? Why pass it as a separate parameter?

File: Source/VCGeneration/Split.cs [hunk 2]
```diff
@@ -985,21 +988,21 @@ bool ContainsAssert(Block b)
...
-              splits.Add(new Split(initialSplit.options, PostProcess(newBlocks), initialSplit.gotoCmdOrigins, initialSplit.parent, initialSplit.Implementation)); // REVIEW: Does gotoCmdOrigins need to be changed at all?
+              splits.Add(new Split(initialSplit.options, PostProcess(newBlocks), initialSplit.gotoCmdOrigins, initialSplit.parent, initialSplit.Implementation, run)); // REVIEW: Does gotoCmdOrigins need to be changed at all?
             }
           }
         }
         return splits;
       }
 
-      public static List<Split> FocusImpl(VCGenOptions options, Implementation impl, Dictionary<TransferCmd, ReturnCmd> gotoCmdOrigins, VCGen par)
+      public static List<Split> FocusImpl(VCGenOptions options, Implementation impl, ImplementationRun run, Dictionary<TransferCmd, ReturnCmd> gotoCmdOrigins, VCGen par)
```
Comment (keyboardDrummer) [inline Source/VCGeneration/Split.cs:1005]: `ImplementationRun` references an `Implementation`, so the `impl` parameter here can be removed.

File: Source/VCGeneration/Split.cs [hunk 3]
```diff
@@ -1120,14 +1123,14 @@ void FocusRec(int focusIdx, IEnumerable<Block> blocks, IEnumerable<Block> freeBl
         return s;
       }
 
-      public static List<Split> FocusAndSplit(VCGenOptions options, Implementation impl, Dictionary<TransferCmd, ReturnCmd> gotoCmdOrigins, VCGen par, bool splitOnEveryAssert)
+      public static List<Split> FocusAndSplit(VCGenOptions options, Implementation impl, ImplementationRun run, Dictionary<TransferCmd, ReturnCmd> gotoCmdOrigins, VCGen par, bool splitOnEveryAssert)
```
Comment (keyboardDrummer) [inline Source/VCGeneration/Split.cs:1126]: `ImplementationRun` references an `Implementation`, so the `impl` parameter here can be removed.

File: Source/VCGeneration/Split.cs [hunk 4]
```diff
@@ -1120,14 +1123,14 @@ void FocusRec(int focusIdx, IEnumerable<Block> blocks, IEnumerable<Block> freeBl
...
       {
-        List<Split> focussedImpl = FocusImpl(options, impl, gotoCmdOrigins, par);
-        var splits = focussedImpl.Select(s => FindManualSplits(s, splitOnEveryAssert)).SelectMany(x => x).ToList();
+        List<Split> focussedImpl = FocusImpl(options, impl, run, gotoCmdOrigins, par);
+        var splits = focussedImpl.Select(s => FindManualSplits(s, run, splitOnEveryAssert)).SelectMany(x => x).ToList();
         return splits;
       }
 
-      public static List<Split /*!*/> /*!*/ DoSplit(Split initial, double splitThreshold, int maxSplits)
+      public static List<Split /*!*/> /*!*/ DoSplit(Split initial, ImplementationRun run, double splitThreshold, int maxSplits)
```
Comment (keyboardDrummer) [inline Source/VCGeneration/Split.cs:1133]: Consider getting the `run` from `Split initial`

**Comment by shazqadeer** [comment]
@keyboardDrummer : With my latest commit, I have replaced the ```implementation``` field in Split with ```run``` and made ```Implementation``` a derived attribute that looks up ```run```. All the other plumbing is the way you suggested.

**Review by keyboardDrummer — APPROVED** [review]

**Pull request merged**